### PR TITLE
Lotka-Volterra

### PR DESCRIPTION
**Pull request by vuolleko**
New example: the stochastic Lotka-Volterra model
+ fix bounds in methods.bo.utils.minimize and shape in visualization.plot_marginals

### CODE REVIEW

File: elfi/examples/lotka_volterra.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    sigma = np.asanyarray(sigma).reshape(-1)
+
+    n_full = 20000
+    stock = np.empty((batch_size, n_full, 2), dtype=np.int32)
+    stock[:, 0, 0] = prey_init
+    stock[:, 0, 1] = predator_init
+    stoichiometry = np.array([[1, 0], [-1, 1], [0, -1], [0, 0]], dtype=np.int32)
+    times = np.empty((batch_size, n_full))
+    times[:, 0] = 0
+    ii = 0
```
Comment (perdaug) [inline elfi/examples/lotka_volterra.py]: Could be moved to just before the start of the while loop to indicate that `ii` is auxiliary

File: elfi/examples/lotka_volterra.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        ii += 1
+
+        # increase the size of arrays if needed
+        if ii == n_full:
+            stock = np.concatenate((stock, np.empty((batch_size, n_full, 2))), axis=1)
+            times = np.concatenate((times, np.empty((batch_size, n_full))), axis=1)
+            n_full *= 2
+
+        # reaction probabilities
+        hazards = np.column_stack((r1 * stock[:, ii-1, 0],
```
Comment (perdaug) [inline elfi/examples/lotka_volterra.py]: My `make lint` complains of `E226 missing whitespace around arithmetic operator`.

File: elfi/examples/lotka_volterra.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    times = times[:, :ii+1]
+
+    times_out = np.linspace(0, time_end, n_obs)
+    stock_out = np.empty((batch_size, n_obs, 2), dtype=np.int32)
+    stock_out[:, 0, :] = stock[:, 0, :]
+
+    # observations at even intervals
+    for ii in range(1, n_obs):
+        iy, ix = np.where(times >= times_out[ii])
+        iy, iix = np.unique(iy, return_index=True)
```
Comment (perdaug) [inline elfi/examples/lotka_volterra.py]: the first `iy` is rewritten by the second one. If the first one is a dummy instance, would suggest calling it `_` instead.

File: elfi/examples/lotka_volterra.py (hunk 2 above)
Comment (vuolleko) [reply elfi/examples/lotka_volterra.py]: Hmm... I think you have some linting package that we don't suggest.

**Pull request merged**